### PR TITLE
BaSP-MR-78: Input component and its logic created

### PR DESCRIPTION
**Pull request by ale-schmidt**
Input component and its logic created

### CODE REVIEW

**Comment by lorenzomauro59** [comment]
[[Shared-Components] Input](https://trello.com/c/S9vSd6nh/78-shared-components-input)

File: src/Components/Shared/Input/index.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  useEffect(() => {
+    if (editMode) {
+      switch (inputType) {
+        case 'text':
+          setText(data);
+          break;
+
+        case 'date':
+          setDate(data);
+          // que el componente haga la transformación o que lo haga el componente hablar con gonza
```
Comment (Federico-Peralta) [inline src/Components/Shared/Input/index.jsx]: You should delete this comment

Comment (Federico-Peralta) [reply src/Components/Shared/Input/index.jsx]: And the others comments

**Comment by lorenzomauro59** [comment]
Everithing looks nice! , only you should add shared buttons

File: src/Components/Shared/Input/index.jsx (hunk 1 above)
Comment (ale-schmidt) [reply src/Components/Shared/Input/index.jsx]: done!

**Review by Federico-Peralta — APPROVED** [review]
Approved!

**Review by lorenzomauro59 — APPROVED** [review]
Looks well!

File: src/Components/Shared/Input/index.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    );
+  };
+
+  const renderInputType = (inputType) => {
+    if (inputType === 'text') {
+      return (
+        <label className={styles['form-label']}>
+          <h2>{labelName}</h2>
+          <input
+            className={styles['form-input']}
```
Comment (EmaBaldor) [inline src/Components/Shared/Input/index.jsx:71]: It may be convenient to also add the 'name' prop.

**Comment by ale-schmidt** [comment]
[[Shared-Components] Input](https://trello.com/c/S9vSd6nh/78-shared-components-input)

**Review by EmaBaldor — APPROVED** [review]

**Review by Gonzarey96 — APPROVED** [review]

**Comment by nazarenapantanetti** [comment]
Tested, the same component has not been applied in all entities yet.

**Pull request merged**